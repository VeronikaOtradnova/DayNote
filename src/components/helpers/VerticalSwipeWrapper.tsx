import { useState } from "react"

interface IProps {
  children: React.ReactNode,
  onTop?: () => void,
  onBottom?: () => void,
}

export function VerticalSwipeWrapper({children, onTop = () => {}, onBottom = () => {}}: IProps) {
  const [touchStart, setTouchStart] = useState<number | null>(null)
  const [touchEnd, setTouchEnd] = useState<number | null>(null)

  // the required distance between touchStart and touchEnd to be detected as a swipe
  const minSwipeDistance = 50

  const onTouchStart = (e: React.TouchEvent<HTMLDivElement>) => {
    setTouchEnd(null)
    setTouchStart(e.targetTouches[0].clientY)
  }

  const onTouchMove = (e: React.TouchEvent<HTMLDivElement>) => {
    setTouchEnd(e.targetTouches[0].clientY);
  };

  const onTouchEnd = () => {
    if (!touchStart || !touchEnd) return
    const distance = touchStart - touchEnd
    const isTopSwipe = distance > minSwipeDistance
    const isBottomSwipe = distance < -minSwipeDistance
    if (isTopSwipe || isBottomSwipe) {
      isTopSwipe ? onTop() : onBottom()
    }
  }

  return (
    <div 
      onTouchStart={onTouchStart} 
      onTouchMove={onTouchMove} 
      onTouchEnd={onTouchEnd} 
    >
      {children}
    </div>
  )
}