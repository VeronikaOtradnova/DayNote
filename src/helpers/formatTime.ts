export function formatTime(timeInMillis: number) {
  const date = new Date(timeInMillis);
  
  // Получаем часы и минуты из объекта Date
  let hours = date.getUTCHours().toString();
  let minutes = date.getUTCMinutes().toString();
  
  // Форматируем часы и минуты в виде двухзначных чисел
  hours = hours.padStart(2, '0');
  minutes = minutes.padStart(2, '0');
  
  // Возвращаем строку в формате 'hh:mm'
  return `${hours}:${minutes}`;
}