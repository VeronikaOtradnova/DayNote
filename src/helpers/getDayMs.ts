export const getDayMs = () => {
  return 24*60*60*1000;
}