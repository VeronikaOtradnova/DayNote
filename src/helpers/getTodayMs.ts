export const getTodayMs = () => {
  return +(new Date().setHours(0, 0, 0, 0));
}