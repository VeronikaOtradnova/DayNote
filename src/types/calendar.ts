export enum calendarActionTypes {
  OPEN_CALENDAR = 'OPEN_CALENDAR',
  CLOSE_CALENDAR = 'CLOSE_CALENDAR',
  SET_CALENDAR_DATE = 'SET_SELECTED_DATE',
}

export interface IOpenCalendarAction {
  type: calendarActionTypes.OPEN_CALENDAR,
}
export interface ICloseCalendarAction {
  type: calendarActionTypes.CLOSE_CALENDAR,
}
export interface ISetCalendarDateAction {
  type: calendarActionTypes.SET_CALENDAR_DATE,
  payload: number,
}

export type TActionType = IOpenCalendarAction | ICloseCalendarAction | ISetCalendarDateAction;

export interface ICalendarState {
  isCalendarOpen: boolean,
  calendarDate: number,
}
