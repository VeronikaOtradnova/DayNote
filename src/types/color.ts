export enum colors {
  green = 'green',
  yellow = 'yellow',
  gray = 'gray',
}

export type TColor = colors.green | colors.yellow | colors.gray;