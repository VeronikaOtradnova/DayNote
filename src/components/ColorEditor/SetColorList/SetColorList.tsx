import { colors } from '../../../types/color';
import { SetColorItem } from '../SetColorItem/SetColorItem';
import styles from './SetColorList.module.css';

interface IProps {
  setEditorOpen?: (value: boolean) => void;
}

export const SetColorList = ({setEditorOpen}: IProps) => {
  return (
    <ul className={styles.list} data-testid="set-color-list">
      <SetColorItem setEditorOpen={setEditorOpen} key={colors.green} color={colors.green} />
      <SetColorItem setEditorOpen={setEditorOpen} key={colors.pink} color={colors.pink} />
      <SetColorItem setEditorOpen={setEditorOpen} key={colors.blue} color={colors.blue} />
      <SetColorItem setEditorOpen={setEditorOpen} key={colors.gray} color={colors.gray} />
    </ul>
  )
}