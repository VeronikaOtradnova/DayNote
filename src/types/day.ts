import { TColor } from "./color";

export interface IDay {
  date: number,
  color: TColor,
}

export enum dayActionTypes {
  ADD_DAY = 'ADD_DAY',
  EDIT_DAY = 'EDIT_DAY',
  SET_CURRENT_DAY = 'SET_CURRENT_DAY'
}

export interface IAddDayAction {
  type: dayActionTypes.ADD_DAY,
  payload: IDay,
}
export interface IEditDayAction {
  type: dayActionTypes.EDIT_DAY,
  payload: IDay,
}
export interface ISetCurrentDayAction {
  type: dayActionTypes.SET_CURRENT_DAY,
  payload: number,
}

export type TDayAction = IAddDayAction | IEditDayAction | ISetCurrentDayAction;

export interface IDayState {
  currentDay: number,
  days: IDay[],
}