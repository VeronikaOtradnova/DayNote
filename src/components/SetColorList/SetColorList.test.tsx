import { render, screen } from "@testing-library/react"
import { SetColorList } from "./SetColorList"
import { colors } from "../../types/color";
import { renderWithRedux } from "../../tests/helpers/renderWithRedux";

describe('TEST SET-COLORS-LIST', () => {
  test('List should render with all colors', () => {
    renderWithRedux(<SetColorList />);
    
    const list = screen.getByTestId('set-color-list');
    expect(list).toBeInTheDocument();
    expect(list).toMatchSnapshot();

    const colorsLength = Object.keys(colors).length;
    expect(screen.getAllByTestId('set-color-item').length).toBe(colorsLength);
  })
})