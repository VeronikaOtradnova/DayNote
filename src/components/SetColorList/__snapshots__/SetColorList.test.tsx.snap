// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST SET-COLORS-LIST List should render with all colors 1`] = `
<ul
  class="list"
  data-testid="set-color-list"
>
  <li
    class="item green"
    data-testid="set-color-item"
  >
    green
  </li>
  <li
    class="item yellow"
    data-testid="set-color-item"
  >
    yellow
  </li>
  <li
    class="item gray"
    data-testid="set-color-item"
  >
    gray
  </li>
</ul>
`;
